mozdownload/scraper.py

```diff
@@ -474,7 +481,11 @@ def get_build_info_for_date(self, date, build_index=None):
             'PLATFORM': '' if self.application not in (
                 'fennec') else '-' + self.platform
         }
-
+        if self.application == 'fenix':
+            regex = r'%(DATE)s-(\d+-){3}fenix-(\d+\.\d.\d)-%(PLATFORM)s$' % {
```

Comment:
Would you mind to create another global mapping for these two regex'es? That would allow us to not have different code paths here.

Otherwise downloading a Fenix build seem to work fine with the other issues from above applied and the binary run with `-t daily -a fenix -p android-x86_64 --date 2022-12-31`. \o/